synapse/rest/client/v1/room.py

```diff
@@ -301,6 +301,11 @@ def on_GET(self, request):
         try:
             yield self.auth.get_user_by_req(request, allow_guest=True)
         except AuthError as e:
+            # option to allow servers in private federations to require auth
```

Comment:
I don't understand this comment. It doesn't seem to explain what we are doing here at all.